RecoLocalCalo/HcalRecProducers/plugins/alpaka/Mahi.dev.cc

```diff
@@ -0,0 +1,1457 @@
+                  for (int i = 0; i < 8; i++) {
+                    for (int j = 0; j < 8; j++)
+                      printf("%f ", AtA(i, j));
+                    printf("\n");
+                  }
+                  printf("Atb\n");
+                  for (int i = 0; i < 8; i++)
+                    printf("%f ", Atb(i));
+                  printf("\n");
+                  printf("result Amplitudes before nnls\n");
+                  for (int i = 0; i < 8; i++)
+                    printf("%f ", resultAmplitudesVector(i));
+                  printf("\n");
+#endif
+
+                  // for fnnls
+                  calo::multifit::MapSymM<float, NPULSES> matrixLForFnnls{shrMatrixLFnnlsStorage};
+
+                  // run fast nnls
+                  calo::multifit::fnnls(AtA,
+                                        Atb,
+                                        resultAmplitudesVector,
+                                        npassive,
+                                        pulseOffsets,
+                                        matrixLForFnnls,
+                                        nnlsThresh,
+                                        nMaxItersNNLS,
+                                        10,
+                                        10);
+
+#ifdef HCAL_MAHI_GPUDEBUG
+                  printf("result Amplitudes\n");
+                  for (int i = 0; i < 8; i++)
+                    printf("resultAmplitudes(%d) = %f\n", i, resultAmplitudesVector(i));
+#endif
+
+                  calo::multifit::calculateChiSq(
+                      matrixL, glbPulseMatrixView, resultAmplitudesVector, inputAmplitudesView, chi2);
+
+                  auto const deltaChi2 = std::abs(chi2 - previous_chi2);
+                  if (chi2 == chi2_2itersback && chi2 < previous_chi2)
+                    break;
+
+                  // update
+                  chi2_2itersback = previous_chi2;
+                  previous_chi2 = chi2;
+
+                  // exit condition
+                  if (deltaChi2 < deltaChi2Threashold)
+                    break;
+                }
+
+#ifdef HCAL_MAHI_GPUDEBUG
+                for (int i = 0; i < NPULSES; i++)
+                  printf("pulseOffsets(%d) = %d outputAmplitudes(%d) = %f\n",
+                         i,
+                         pulseOffsets(i),
+                         i,
+                         resultAmplitudesVector(i));
+                printf("chi2 = %f\n", chi2);
+#endif
+
+                outputGPU.chi2()[gch] = chi2;
+                auto const idx_for_energy = std::abs(pulseOffsetsView.offsets()[0]);
+                outputGPU.energy()[gch] = (gain * resultAmplitudesVector(idx_for_energy)) * respCorrection;
+
+              }  // loop over channels
+            }    //loop over group of channels
+          }
+        };  // Kernel_minimize
+
+      }  // namespace mahi
+    }    // namespace hcal::reconstruction
+    namespace hcal {
+      namespace reconstruction {
```

Comment:
actually, just
```suggestion
```

Reply:
done in [3bed955](https://github.com/cms-sw/cmssw/pull/44910/commits/3bed955698f044472ce6896e6dc91d35a17c0393)